jdbcSource/src/main/java/io/vantiq/extsrc/jdbcSource/JDBCHandleConfiguration.java

```diff
@@ -214,6 +208,107 @@ public void run() {
         log.trace("JDBC source created");
         return true;
     }
+
+    /**
+     * Method used to create the query and publish handlers
+     * @param generalConfig     The general configuration of the JDBC Source
+     * @return                  Returns the maximum pool size, equal to twice the number of active tasks.
+     *                          If default active tasks is used, then returns 0.
+     */
+    private int createQueryAndPublishHandlers(Map<String, ?> generalConfig) {
+        int maxPoolSize = 0;
+
+        // Checking if asynchronous processing was specified in the general configuration
+        if (generalConfig.get(ASYNCH_PROCESSING) instanceof Boolean && (Boolean) generalConfig.get(ASYNCH_PROCESSING)) {
+            asynchronousProcessing = true;
+            int maxActiveTasks = MAX_ACTIVE_TASKS;
+            int maxQueuedTasks = MAX_QUEUED_TASKS;
+
+            if (generalConfig.get(MAX_ACTIVE) instanceof Integer && (Integer) generalConfig.get(MAX_ACTIVE) > 0) {
+                maxActiveTasks = (Integer) generalConfig.get(MAX_ACTIVE);
+
+                // Used to set the max pool size for connection pool
+                maxPoolSize = 2*maxActiveTasks;
+            }
+
+            if (generalConfig.get(MAX_QUEUED) instanceof Integer && (Integer) generalConfig.get(MAX_QUEUED) > 0) {
+                maxQueuedTasks = (Integer) generalConfig.get(MAX_QUEUED);
+            }
+
+            // Creating the thread pool executors with Queue
+            source.queryPool = new ThreadPoolExecutor(maxActiveTasks, maxActiveTasks, 0l, TimeUnit.MILLISECONDS,
+                    new LinkedBlockingQueue<Runnable>(maxQueuedTasks));
+            source.publishPool = new ThreadPoolExecutor(maxActiveTasks, maxActiveTasks, 0l, TimeUnit.MILLISECONDS,
+                    new LinkedBlockingQueue<Runnable>(maxQueuedTasks));
```

Comment:
We can add the no `LinkedBlockingQueue` case when `maxQueuedTasks` is zero later.